### ISSUE TITLE
No Webmention Support with Indirect Source Link In Post

### ISSUE THREAD

**Issue by slyduda** [snarfed/bridgy#1322]
TLDR: I believe the bridgy app is associating the wrong link to my actual post due to a successful server side redirect, but an unsuccessful client side redirect

This is a bit of an odd issue because I believe I am trying to do something with links in posts that was not intended, so I will try to explain this in a bit more depth:

- I have a static website at https://slyduda.com
- In an effort to shorten links for the sake of social media length limits I bought the domain https://sly.re
- I have a double redirect that goes from sly.re/[id-of-post] -> slyduda.com/p/[id-of-post] -> slyduda.com/post/[SLUG-of-post]
- The first redirect is a 301 directly from nginx, but the second redirect is from html headers <meta http-equiv="refresh" content="0;URL='https://slyduda.com/post/[SLUG-of-post]">
- For clarity, I am using the [following for a client-side redirect](https://www.w3.org/TR/WCAG20-TECHS/H76.html)

As an example I have this successful Bridgy Twitter post: https://twitter.com/i/web/status/1581159763227787264

The last link is not a bridgy link, but a custom link that my static blog generates and makes invisible to the readers in the e-content of my website post located here: https://slyduda.com/post/7c9255e6-832e-4632-b7a2-978f30a9046b/

Clicking the link in the post will take the user from: https://sly.re/4thf0g -> https://slyduda.com/p/4thf0g -> https://slyduda.com/post/7c9255e6-832e-4632-b7a2-978f30a9046b/

[This part of the Bridgy docs](https://brid.gy/about#missing) explains why syndication might be broken. Along with that in my Bridgy dashboard the responses show `No webmention support for slyduda.com/p/4thf0g` which tells me that it successfully redirected from point A to B but not B to C

I am under the assumption that Bridgy is unable to take into account a client side redirect (probably for good reason), but I am still wondering if what I have explained is accurate and if there are any alternative solutions!

If not, no worries, and thank you for the help in advance!

**Comment by snarfed** [comment]
Hi! Thanks for all the details. You're right, the problem here is probably that Bridgy doesn't support client-side redirects: https://github.com/snarfed/bridgy/issues/253. It shouldn't be too hard to add that support though, I'd happily accept PR(s)!

**Comment by snarfed** [comment]
Oh, btw, you might consider deprioritizing your URL shortening. Of the major social networks that limit post length, eg Twitter and Mastodon, they all switched to counting URLs at a fixed number of characters, eg 23 for Twitter, a long time ago. None that I'm aware of count the full length of URLs against you any more.

More background:

- https://indieweb.org/permashortlink#Why ("Older reasons that apply less or not at all nowadays")
- https://indieweb.org/shortlink#Criticism
- https://indieweb.org/URL_design#Avoid (note that char counting isn't one of the reasons)

**Comment by slyduda** [comment]
Awesome! I will do some work on this today and tomorrow then :)

The fixed link character is new information to me and I appreciate the resources. If it becomes too difficult to implement elegantly I will de-prioritize it like you are suggesting.

**Comment by snarfed** [comment]
Great! You'll probably want to add it here:

https://github.com/snarfed/webutil/blob/9ed4fb34e1371f38374ba0bc1f5647a48090bb01/util.py#L1970-L1988

New test(s) would go into [`test_util.py`](https://github.com/snarfed/webutil/blob/main/tests/test_util.py).

**Comment by snarfed** [comment]
Merging into #253 as a dupe.
